### ISSUE TITLE
`Server initialization failed` (`ModuleNotFoundError: No module named 'packaging'`) when using bundled Mypy

### ISSUE THREAD

**Issue by posita** [microsoft/vscode-mypy#148]
Observed on pre-release `v2023.3.12551007`.

### Steps

* One of:
  * On Linux (e.g., Ubuntu 20.04.6 with Python installed from `ppa:deadsnakes/ppa`; see below)
    * `Python: Select Interpreter` -> `/usr/bin/python3.10`
    * `Python: Select Interpreter` -> `/usr/bin/python3.11`
  * On Mac
    * `Python: Select Interpreter` -> `/usr/local/bin/python3.10` (installed from [`python-3.10.11-macos11.pkg`](https://www.python.org/ftp/python/3.10.11/python-3.10.11-macos11.pkg))
    * `Python: Select Interpreter` -> `/usr/local/bin/python3.11` (installed from [`python-3.11.5-macos11.pkg`](https://www.python.org/ftp/python/3.11.5/python-3.11.5-macos11.pkg))
* Open a Python source file and add `i: int = "1"`

### Results

```
2023-09-14 18:36:07.066 [info] Traceback (most recent call last):

2023-09-14 18:36:07.066 [info]   File "/home/.../.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/tool/lsp_server.py", line 18, in <module>

2023-09-14 18:36:07.066 [info]     from packaging.version import Version

2023-09-14 18:36:07.067 [info] ModuleNotFoundError: No module named 'packaging'

2023-09-14 18:36:07.076 [info] [Error - 6:36:07 PM] Server initialization failed.
2023-09-14 18:36:07.076 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 18:36:07.076 [info] [Error - 6:36:07 PM] The Mypy Type Checker server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
2023-09-14 18:36:07.076 [info] [Error - 6:36:07 PM] Mypy Type Checker client: couldn't create connection to server.
2023-09-14 18:36:07.076 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 18:36:07.076 [info] [Error - 6:36:07 PM] Restarting server failed
2023-09-14 18:36:07.076 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
```

Switching the Python interpreter to one that lives in a virtual env and that has Mypy installed (e.g., via `/path/to/venv/bin/pip install mypy`) corrects the above problem. It seems like the bundled version is missing prerequisites?

---

Python installation on Ubuntu was as follows:

```
sudo add-apt-repository --yes ppa:deadsnakes/ppa
sudo apt-get update
sudo apt-get install --yes \
  python3.10 \
  python3.10-dev \
  python3.10-venv \
  python3.11 \
  python3.11-dev \
  python3.11-venv
```

**Comment by karthiknadig** [comment]
@posita Can you share the initial part of the logs with `sys.path`? it should include a path to `bundled\libs`, if it does can you check that location to see if it contains `packaging`?

**Comment by karthiknadig** [comment]
Also, what do you have for `mypy-type-checker.importStrategy`?

**Comment by posita** [comment]
`mypy-type-checker.importStrategy` is unset (i.e., the default of `useBundled`).

`bundled/libs` (`packaging` exists there):

``` sh
% ls ~/.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs
total 232
  4 __pycache__/                4 exceptiongroup-1.1.2.dist-info/        8 mypy_extensions.py           4 typeguard/
  4 attr/                       4 importlib_metadata/                    4 mypyc/                       4 typeguard-3.0.2.dist-info/
  4 attrs/                      4 importlib_metadata-6.8.0.dist-info/    4 packaging/                   4 typing_extensions-4.7.1.dist-info/
  4 attrs-23.1.0.dist-info/     4 lsprotocol/                            4 packaging-23.1.dist-info/  112 typing_extensions.py
  4 cattr/                      4 lsprotocol-2023.0.0a2.dist-info/       4 pygls/                       4 zipp/
  4 cattrs/                     4 mypy/                                  4 pygls-1.0.2.dist-info/       4 zipp-3.16.2.dist-info/
  4 cattrs-23.1.2.dist-info/    4 mypy-1.5.0.dist-info/                  4 tomli/
  4 exceptiongroup/             4 mypy_extensions-1.0.0.dist-info/       4 tomli-2.0.1.dist-info/
% ( cd "${HOME}" && find .vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging* )
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/markers.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/py.typed
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/_tokenizer.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/_manylinux.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/_elffile.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/version.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/__init__.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/_structures.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/tags.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/_parser.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/utils.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/requirements.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/specifiers.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/_musllinux.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging/metadata.py
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging-23.1.dist-info
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging-23.1.dist-info/LICENSE
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging-23.1.dist-info/RECORD
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging-23.1.dist-info/LICENSE.BSD
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging-23.1.dist-info/LICENSE.APACHE
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging-23.1.dist-info/METADATA
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging-23.1.dist-info/INSTALLER
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging-23.1.dist-info/WHEEL
.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs/packaging-23.1.dist-info/REQUESTED
% /usr/bin/python3.11 ~/.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/tool/lsp_server.py
Traceback (most recent call last):
  File "/home/.../.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/tool/lsp_server.py", line 18, in <module>
    from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'
% PYTHONPATH="${HOME}/.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/libs" /usr/bin/python3.11 ~/.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/tool/lsp_server.py
<hangs here; seems to work?>
```

Extended logs:

```
2023-09-14 19:28:05.081 [info] No interpreter found from setting mypy-type-checker.interpreter
2023-09-14 19:28:05.081 [info] Getting interpreter from ms-python.python extension for workspace /...
2023-09-14 19:28:05.085 [info] Interpreter from ms-python.python extension for /...: /usr/bin/python3.11
2023-09-14 19:28:05.091 [info] Server run command: /usr/bin/python3.11 /home/.../.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/tool/lsp_server.py
2023-09-14 19:28:05.092 [info] Server: Start requested.
2023-09-14 19:28:05.171 [info] Traceback (most recent call last):
  File "/home/.../.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/tool/lsp_server.py", line 18, in <module>

2023-09-14 19:28:05.171 [info]     from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'

2023-09-14 19:28:05.180 [info] [Error - 7:28:05 PM] Server initialization failed.
2023-09-14 19:28:05.180 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.180 [info] [Info  - 7:28:05 PM] Connection to server got closed. Server will restart.
2023-09-14 19:28:05.180 [info] true
2023-09-14 19:28:05.180 [info] [Error - 7:28:05 PM] Mypy Type Checker client: couldn't create connection to server.
2023-09-14 19:28:05.180 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.254 [info] Traceback (most recent call last):
  File "/home/.../.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/tool/lsp_server.py", line 18, in <module>

2023-09-14 19:28:05.254 [info]     from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'

2023-09-14 19:28:05.262 [info] [Error - 7:28:05 PM] Server initialization failed.
2023-09-14 19:28:05.262 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.262 [info] [Info  - 7:28:05 PM] Connection to server got closed. Server will restart.
2023-09-14 19:28:05.262 [info] true
2023-09-14 19:28:05.262 [info] [Error - 7:28:05 PM] Mypy Type Checker client: couldn't create connection to server.
2023-09-14 19:28:05.262 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.263 [error] Server: Start failed: Error: Pending response rejected since connection got disposed
2023-09-14 19:28:05.365 [info] Traceback (most recent call last):
  File "/home/.../.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/tool/lsp_server.py", line 18, in <module>
    from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'

2023-09-14 19:28:05.366 [info] [Error - 7:28:05 PM] Server initialization failed.
2023-09-14 19:28:05.367 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.367 [info] [Info  - 7:28:05 PM] Connection to server got closed. Server will restart.
2023-09-14 19:28:05.367 [info] true
2023-09-14 19:28:05.367 [info] [Error - 7:28:05 PM] Mypy Type Checker client: couldn't create connection to server.
2023-09-14 19:28:05.367 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.367 [info] [Error - 7:28:05 PM] Restarting server failed
2023-09-14 19:28:05.367 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.446 [info] Traceback (most recent call last):
  File "/home/.../.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/tool/lsp_server.py", line 18, in <module>
    from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'

2023-09-14 19:28:05.454 [info] [Error - 7:28:05 PM] Server initialization failed.
2023-09-14 19:28:05.454 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.454 [info] [Info  - 7:28:05 PM] Connection to server got closed. Server will restart.
2023-09-14 19:28:05.454 [info] true
2023-09-14 19:28:05.454 [info] [Error - 7:28:05 PM] Mypy Type Checker client: couldn't create connection to server.
2023-09-14 19:28:05.454 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.454 [info] [Error - 7:28:05 PM] Restarting server failed
2023-09-14 19:28:05.454 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.527 [info] Traceback (most recent call last):
  File "/home/.../.vscode-server/extensions/ms-python.mypy-type-checker-2023.3.12551007/bundled/tool/lsp_server.py", line 18, in <module>

2023-09-14 19:28:05.527 [info]     from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'

2023-09-14 19:28:05.534 [info] [Error - 7:28:05 PM] Server initialization failed.
2023-09-14 19:28:05.534 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.534 [info] [Error - 7:28:05 PM] The Mypy Type Checker server crashed 5 times in the last 3 minutes. The server will not be restarted. See the output for more information.
2023-09-14 19:28:05.534 [info] [Error - 7:28:05 PM] Mypy Type Checker client: couldn't create connection to server.
2023-09-14 19:28:05.534 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
2023-09-14 19:28:05.534 [info] [Error - 7:28:05 PM] Restarting server failed
2023-09-14 19:28:05.534 [info]   Message: Pending response rejected since connection got disposed
  Code: -32097 
```

**Comment by posita** [comment]
I think I found the issue (fix in #151).
